tools/topology/sof-glk-da7219-kwd.m4

```diff
@@ -14,18 +15,64 @@ include(`common/tlv.m4')
 # Include Token library
 include(`sof/tokens.m4')
 
-# Include bxt DSP configuration
-include(`platform/intel/bxt.m4')
+ifelse(PLATFORM, `bxt', `# Include bxt DSP configuration', `')
+ifelse(PLATFORM, `cml', `# Include cnl DSP configuration', `')
+
+ifelse(PLATFORM, `bxt', `include(`platform/intel/bxt.m4')', `')
+ifelse(PLATFORM, `cml', `include(`platform/intel/cnl.m4')', `')
+
+ifelse(PLATFORM, `cml', `include(`abi.h')', `')
+
 include(`platform/intel/dmic.m4')
 
 define(KWD_PIPE_SCH_DEADLINE_US, 20000)
 
+undefine(`SSP_INDEX')
+define(`SSP_INDEX', ifelse(PLATFORM, `bxt', `2',
+	ifelse(PLATFORM, `cml', `0', `')))
+
+undefine(`SSP_NAME')
+define(`SSP_NAME', ifelse(PLATFORM, `bxt', `SSP2-Codec',
+	ifelse(PLATFORM, `cml', `SSP0-Codec', `')))
+
+undefine(`DMIC0_PIPE_NAME')
+define(`DMIC0_PIPE_NAME', ifelse(PLATFORM, `bxt', `volume',
+	ifelse(PLATFORM, `cml', `passthrough', `')))
+
+undefine(`DMIC_PCM_NUM')
+define(`DMIC_PCM_NUM', ifelse(PLATFORM, `bxt', `99',
+	ifelse(PLATFORM, `cml', `2', `')))
+
+undefine(`DMIC01_FMT')
+define(`DMIC01_FMT', ifelse(PLATFORM, `bxt', `s16le',
+	ifelse(PLATFORM, `cml', `s32le', `')))
+
+undefine(`DMIC1_FMT')
+define(`DMIC1_FMT', ifelse(PLATFORM, `bxt', `s16le',
+	ifelse(PLATFORM, `cml', `s24le', `')))
+
+undefine(`AMP_DAI_FMT')
+define(`AMP_DAI_FMT', ifelse(PLATFORM, `bxt', `s16le',
+	ifelse(PLATFORM, `cml', `s24le', `')))
+
+undefine(`HDMI0_INDEX')
+define(`HDMI0_INDEX', ifelse(PLATFORM, `bxt', `3',
+	ifelse(PLATFORM, `cml', `0', `')))
+
+undefine(`HDMI1_INDEX')
+define(`HDMI1_INDEX', ifelse(PLATFORM, `bxt', `4',
+	ifelse(PLATFORM, `cml', `1', `')))
+
+undefine(`HDMI2_INDEX')
+define(`HDMI2_INDEX', ifelse(PLATFORM, `bxt', `5',
+	ifelse(PLATFORM, `cml', `2', `')))
```

Comment:
This growth in conditional logic defeats the purpose of the include/define directives. Would we do this in a C file? No.
Far better to have a cml-blah-config and bxt-blah_config.m4 and inlude the correct file (with all these macro defs for format etc.)

Reply:
@macchian please align with @fredoh9 as he is working on the same fix for current upstream topologies. This can be applied after his update.

Reply:
Thanks @lgirdwood. @fredoh9 , could you guide me how to include the marcor defs by separate m4 file? Or existed same code can share? Thanks.

Reply:
@macchian see https://github.com/thesofproject/sof/pull/2444 and https://github.com/thesofproject/sof/pull/2461 I think @fredoh9 may have an update later today.

Reply:
I will update them today

Reply:
Thanks @fredoh9 . I followed your codes by included cml.m4. Rebased and submitted #2497 for reviewing.